eventcrypto.go

```diff
@@ -0,0 +1,161 @@
+package gomatrixserverlib
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"golang.org/x/crypto/ed25519"
+)
+
+// AddContentHashesToEvent sets the "hashes" key of the event with a SHA-256 hash of the unredacted event content.
+// This hash is used to detect whether the unredacted content of the event is valid.
```

Comment:
Would be nice to state that you're returning the event with said hashes somewhere, it's all rather implicit atm and you can't use the types to infer it since it's just `[]byte]`.